frontend/src/pages/MedDog.tsx

```diff
@@ -1,31 +1,82 @@
 import './MedDog.scss';
-import {LocalizationProvider, StaticDatePicker} from "@mui/lab";
-import AdapterDateFns from '@mui/lab/AdapterDateFns';
-import {TextField} from "@mui/material";
-import React from "react";
+import React, {ChangeEvent, useContext, useState} from "react";
+import AppointmentGallery from "../components/AppointmentGallery";
+import AddCircleOutlineIcon from '@mui/icons-material/AddCircleOutline';
+import {Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField} from "@mui/material";
+import {DateTimePicker, LocalizationProvider} from "@mui/lab";
+import AdapterDateFns from "@mui/lab/AdapterDateFns";
+import {addAppointment} from "../service/RequestService";
+import {AuthContext} from "../context/AuthProvider";
 
 export default function MedDog() {
 
-    const [value, setValue] = React.useState<Date | null>(new Date());
+    const [open, setOpen] = React.useState(false);
+    const [appointment, setAppointment] = useState<string>("")
+    const [dateAndTime, setDateAndTime] = React.useState<Date | null>(new Date());
+
+    const {token} = useContext(AuthContext)
+
+    const handleClickOpen = () => {
```

Comment:
Does this function need to know that it is called on click action?

Reply:
You mean i should rename the function to just 'handleOpen'?